app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java

```diff
@@ -340,23 +341,44 @@ private void startDecodeQrCodeImages(List<Uri> uris) {
         QrDecodeTask task = new QrDecodeTask(this, (results) -> {
             List<CharSequence> errors = new ArrayList<>();
             List<VaultEntry> entries = new ArrayList<>();
+            List<GoogleAuthInfo.Export> googleAuthExports = new ArrayList<>();
+
             for (QrDecodeTask.Result res : results) {
                 if (res.getException() != null) {
                     errors.add(buildImportError(res.getFileName(), res.getException()));
                     continue;
                 }
 
                 try {
-                    GoogleAuthInfo info = GoogleAuthInfo.parseUri(res.getResult().getText());
-                    VaultEntry entry = new VaultEntry(info);
-                    entries.add(entry);
-                } catch (GoogleAuthInfoException e) {
+                    Uri scanned = Uri.parse(res.getResult().getText());
+                    if (Objects.equals(GoogleAuthInfo.getUriScheme(scanned), GoogleAuthInfo.SCHEME_EXPORT)) {
+                        GoogleAuthInfo.Export export = GoogleAuthInfo.parseExportUri(scanned);
+                        for (GoogleAuthInfo info: export.getEntries()) {
+                            VaultEntry entry = new VaultEntry(info);
+                            entries.add(entry);
+                        }
+                        googleAuthExports.add(export);
+                    } else {
+                        GoogleAuthInfo info = GoogleAuthInfo.parseUri(res.getResult().getText());
+                        VaultEntry entry = new VaultEntry(info);
+                        entries.add(entry);
+                    }
+                } catch (GoogleAuthInfoException | IllegalArgumentException e) {
                     errors.add(buildImportError(res.getFileName(), e));
                 }
             }
 
             final DialogInterface.OnClickListener dialogDismissHandler = (dialog, which) -> importScannedEntries(entries);
-            if ((errors.size() > 0 && results.size() > 1) || errors.size() > 1) {
+            if (!googleAuthExports.isEmpty()) {
+                try {
+                    List<Integer> missingIndexes = GoogleAuthInfo.Export.getMissingIndexes(googleAuthExports);
+                    if (missingIndexes.size() != 0) {
```

Comment:
Small oversight here. If there are no missing QR codes, we don't import anything.